onnx/onnx.proto

```diff
@@ -53,27 +53,49 @@ enum Version {
   // control. We should use version as
   //     xx(major) - xx(minor) - xxxx(bugfix)
   // and we are starting with 00000001.
-  IR_VERSION = 00000001;
+  IR_VERSION = 00000101;
+  IR_VERSION_2017_10_10 = 00000001;
 }
 
+
+
 // A named attribute containing either singular float, integer, string
 // and tensor values, or repeated float, integer, string and tensor values.
 // An AttributeProto MUST contain the name field, and *only one* of the
 // following content fields, effectively enforcing a C/C++ union equivalent.
+
+enum AttributeType {
```

Comment:
there's already https://github.com/onnx/onnx/blob/b8ab0ed286ffd0b0020a367e078f66bdc3303db5/onnx/defs/schema.h#L151 - maybe use the enum from here and update the library code?

Reply:
Good catch. 

I just changed the proto definition to have the same enum values as schema.h for now. 

The simplest way to actually do this right is to change:

    enum class AttrType {

to 

    typedef <identifier-generated-by-protoc> AttrType;

Let me know if you have a better idea.  I'd rather not scrub all of the call sites to pull on the pb.h identifier directly.

Reply:
Actually, I looked at doing the above, and it's pretty grotesque.

The way enum identifiers get mangled by protoc for both Python and C++ is suboptimal.

To get reasonable Python bindings, I moved the enum to nest inside of AttributeProto. Otherwise, we wind up with top-level identifiers like FLOAT and TENSORS, which feels pretty bad.

However, the generated C++ is pretty bad either way.

If I don't nest AttributeType, then the generated C++ looks like this:

    enum AttributeType {
      AttributeType_FLOAT = 0,
      AttributeType_INT = 1,
      AttributeType_STRING = 2,
     ...

If we do nest AttributeType to make the python language binding reasonable, then the generated C++ looks like this:

    enum AttributeProto_AttributeType {
      AttributeProto_AttributeType_FLOAT = 0,
      AttributeProto_AttributeType_INT = 1,
      AttributeProto_AttributeType_STRING = 2,
     ...

Either way, I'd need to touch a lot of lines of C++, most of which is relatively immune from the proto file.

To that end, I'll add a warning in both files indicating that they should be revved in sync and NOT try to have a single definition.